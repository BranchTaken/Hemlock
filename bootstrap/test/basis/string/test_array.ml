open! Basis.Rudiments
open! Basis
open String
open Format

let test () =
  let test_array a = begin
    let s = of_array a in
    let a' = to_array s in
    let s' = of_array a' in
    printf "array: %a -> ... -> %a\n" xpp s xpp s';
  end in
  test_array [||];
  test_array (Array.of_list [
    (Codepoint.of_char 'a');
  ]);
  test_array (Array.of_list [
    (Codepoint.of_char 'a');
    (Codepoint.of_char 'b');
  ]);
  test_array (Array.of_list [
    (Codepoint.of_char 'a');
    (Codepoint.of_char 'b');
    (Codepoint.of_char 'c');
  ]);
  test_array (Array.of_list [
    (Codepoint.of_char 'a');
    (Codepoint.of_char 'b');
    (Codepoint.of_char 'c');
    (Codepoint.of_char 'd');
  ]);
  test_array (Array.of_list [
    (Codepoint.of_char 'a');
    (Codepoint.of_char 'b');
    (Codepoint.of_char 'c');
    (Codepoint.of_char 'd');
    (Codepoint.of_char 'e');
  ])

let _ = test ()
