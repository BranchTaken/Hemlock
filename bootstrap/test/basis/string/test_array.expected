array: "" -> ... -> ""
array: "a" -> ... -> "a"
array: "ab" -> ... -> "ab"
array: "abc" -> ... -> "abc"
array: "abcd" -> ... -> "abcd"
array: "abcde" -> ... -> "abcde"
